### PR TITLE
removed mutibility.

### PR DESCRIPTION
**Pull request by Christian-B**
fixes https://github.com/SpiNNakerManchester/JavaSpiNNaker/issues/194

### CODE REVIEW

File: SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/RoutingEntry.java [hunk 1]
```diff
@@ -59,8 +59,19 @@ public RoutingEntry(int route) {
...
+            if (procId >= MAX_NUM_CORES || procId < 0) {
+                throw new IllegalArgumentException(
+                    "Processor IDs must be between 0 and "
+                            + (MAX_NUM_CORES - 1) + " found " + procId);
+            }
+            if (!this.processorIDs.contains(procId)) {
+                this.processorIDs.add(procId);
+            }
+        }
+        for (Direction linkIds: linkIDs) {
```
Comment (dkfellows) [inline SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/RoutingEntry.java]: Would it be better to make the collection of directions (and possibly of processors) into something other than a plain `ArrayList`? For example, a `LinkedHashSet` or a `TreeSet`?

Comment (alan-stokes) [reply SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/RoutingEntry.java]: the only thing i can see which might affect that decision, is the two apis for getProcdessorIDs and getLinkIDs. These both expect a List<Direction> and List<Integer>. So converting between a ArrayList and List is a bit trivial. LinkedHashSet or TreeSet...... maybe not so trivial. 

inside the class, its only ever adding, or iterating over the lists. so maybe its not worth changing?

**Comment by alan-stokes** [comment]
so from my point of view, this is fine to merge. leaving for @dkfellows  to decide if hes happy

File: SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/RoutingEntry.java (hunk 1 above)
Comment (dkfellows) [reply SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/RoutingEntry.java]: Hmm. Thinking about this, the stuff I'm thinking about should be a separate change.

**Pull request merged**